src/javascript/ContentEditor/DesignSystem/OrderableValue/OrderableValue.jsx

```diff
@@ -1,71 +1,72 @@
-        type: `REFERENCE_CARD_${field.name}`,
-        item: {name: name},
-        collect: monitor => ({
-            isDragging: monitor.isDragging()
-        })
-    });
+    const [{handlerId}, drop] = useReorderDrop(
+        {ref, index, onReorder: onValueReorder},
+        {
+            accept: `REFERENCE_CARD_${field.name}`
+        });
+    const [{isDragging}, drag, dragPreview] = useReorderDrag(
+        {item: {id, index}, onDrop: onValueReorderDropped, onAbort: onValueReorderAborted},
+        {
+            type: `REFERENCE_CARD_${field.name}`,
+            canDrag: () => isDraggable
+        });
+
+    if (isReferenceCard) {
+        drag(drop(ref));
+    } else {
+        drop(dragPreview(ref));
+    }
 
     return (
-        <div key={name}
-             ref={field.readOnly ? null : drop}
-             id={name}
-             className={styles.fieldComponentContainer}
-             data-sel-content-editor-multiple-generic-field={name}
-             data-sel-content-editor-field-readonly={field.readOnly}
-        >
-            <div className={`${styles.referenceDropGhostHidden} ${isDropping ? styles.referenceDropGhost : ''}`} data-droppable-zone={name}/>
-            {component &&
-                <div ref={isReferenceCard && isDraggable ? drag : null} className={styles.draggableCard}>
-                    {!isDragging &&
-                        <>
-                            {!isReferenceCard &&
-                            <div ref={isDraggable ? drag : null} className={clsx(isDraggable ? styles.draggableIcon : styles.notDraggableIcon)}>
-                                <HandleDrag size="big"/>
-                            </div>}
-                            {component}
-                        </>}
-                    {!isDragging &&
-                    <Tooltip label={t('jcontent:label.contentEditor.edit.fields.actions.clear')}>
-                        <Button variant="ghost"
-                                data-sel-action={`removeField_${index}`}
-                                aria-label={t('jcontent:label.contentEditor.edit.fields.actions.clear')}
-                                icon={<Close/>}
-                                onClick={() => onFieldRemove(index)}
-                        />
-                    </Tooltip>}
-                </div>}
+        <div className={styles.fieldComponentContainer}>
+            <div
+                ref={field.readOnly ? undefined : ref}
+                className={clsx(
+                    styles.draggableCard,
+                    isDragging && styles.draggingCard
+                )}
+                data-sel-content-editor-multiple-generic-field={name}
+                data-sel-content-editor-field-readonly={field.readOnly}
+                data-handler-id={handlerId}
+            >
+                {!isReferenceCard &&
+                    <div ref={isDraggable ? drag : null} className={clsx(isDraggable ? styles.draggableIcon : styles.notDraggableIcon)}>
```

Comment:
just want to confirm change in logic here, as it looks like we now lose drag ref when it's draggable and isReferenceCard